### PR TITLE
Update wrapped implementation

### LINKED ISSUE #26: How about upgrade lucas-clemente/quic-go to v0.14.0
I see currently we used quic-go's version is v0.7.1-0.20190401152353-907071221cf9, how about upgrading it to the latest release version.

### PR DESCRIPTION
**Pull request by cryptix**
TODO:
* [x] update quic-go to v0.17.1 (closes #8)
* [x] added missing NextProtos on tls.Config and context on accept.
* [x] `quic.Session.Close` is gone. seems like it _should_ be some where but i couldn't find it just yet.
* [x] read loop doesn't get unblocked
* [x] fix close handling of streams (166a0ed)
* [?] fix nil panic in pion/webrtc E2E test because b.log isn't initialized (4315de1)

also fixes #26

### CODE REVIEW

**Comment by cryptix** [comment]
I replaced the missing `Close()` with the available `CloseWithError(io.EOF)` and added a test that makes sure the wrapper works as intended (all the data is transported from A to B).

I think this is good for review now.

**Comment by cryptix** [comment]
@Sean-Der not sure who else would review this, that's why i picked you.

ps: the tests pass on travis but something about [the coverage report generation seems broken](https://travis-ci.org/github/pion/quic/jobs/707336136#L789).

**Comment by cryptix** [comment]
Also fixes #33

File: transportbase.go [hunk 1]
```diff
@@ -111,7 +113,7 @@ func (b *TransportBase) acceptStreams() {
 	for {
 		s, err := b.session.AcceptStream()
 		if err != nil {
-			b.log.Errorf("Failed to accept stream: %v", err)
+			fmt.Fprintf(os.Stderr, "Failed to accept stream: %T %v\n", err, err)
```
Comment (Sean-Der) [inline transportbase.go:116]: I would use `b.log.Errorf` if possible. Users can control this with by setting env variables [like this](https://github.com/pion/webrtc/wiki/Debugging-WebRTC) OR they can redirect to a custom log file like [this](https://github.com/pion/webrtc/tree/master/examples/custom-logger)

File: transport_go_test.go [hunk 2]
```diff
@@ -97,15 +148,22 @@ func TestTransport_E2E(t *testing.T) {
...
-func readLoop(s *BidirectionalStream) {
+func readLoop(s *BidirectionalStream, buf io.Writer, done chan<- struct{}) {
+	var bufSz = 1024
+	buffer := make([]byte, bufSz)
 	for {
-		buffer := make([]byte, 15)
-		_, err := s.ReadInto(buffer)
-		if err != nil {
+		res, err := s.ReadInto(buffer)
+		_, _ = buf.Write(buffer[:res.Amount])
```
Comment (Sean-Der) [inline transport_go_test.go:161]: Maybe just do an `assert.NoError(t, err)`

Even if an error is never returned, just nice to not worry about.

File: transport.go [hunk 3]
```diff
@@ -42,15 +45,16 @@ func newServer(url string, config *Config) (*Transport, error) {
...
 	l, err := wrapper.Listen(url, cfg)
 	if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 
 	s, err := l.Accept()
 	if err != nil {
-		return nil, err
+		_ = l.Close()
```
Comment (Sean-Der) [inline transport.go:53]: worth capturing this error and doing a `Wrap` maybe?

File: transportbase.go (hunk 1 above)
Comment (cryptix) [reply transportbase.go:116]: Thanks! I was looking for that and will change it back.

The testsuite setup somehow didn't set it which lead to a panic.

**Pull request merged**